classes/table/files_table.php

```diff
@@ -0,0 +1,101 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Missing files table.
+ *
+ * @package   tool_objectfs
+ * @author    Dmitrii Metelkin <dmitriim@catalyst-au.net>
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_objectfs\table;
+
+defined('MOODLE_INTERNAL') || die();
+
+class files_table extends \table_sql {
```

Comment:
Getting 

```
moodle_1          | 2018/06/07 02:56:36 [error] 37#37: *206 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Class 'table_sql' not found in /siteroot/admin/tool/objectfs/classes/table/files_table.php on line 30
moodle_1          | PHP message: PHP Stack trace:
moodle_1          | PHP message: PHP   1. {main}() /siteroot/admin/tool/objectfs/missing_files.php:0
moodle_1          | PHP message: PHP   2. spl_autoload_call() /siteroot/admin/tool/objectfs/missing_files.php:37
moodle_1          | PHP message: PHP   3. core_component::classloader() /siteroot/admin/tool/objectfs/missing_files.php:37
```

On mdl34 - what version were you running this on ?

Adding 'require_once($CFG->libdir . '/tablelib.php'); ' at the top fixed this

Reply:
Works ok on 3.1